### PR TITLE
openshift_logging link pull secret to serviceaccounts

### PR DESCRIPTION
**Pull request by jcantrill**
This PR:

* Allows a user to provide a pull-secret to link to the logging service accounts
* fixes an error when undeploying logging

cc @ewolinetz @sdodson 

aos-ci-test

### CODE REVIEW

**Comment by abutcher** [comment]
aos-ci-test

File: roles/openshift_logging/tasks/oc_secret.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+- command: >
```
Comment (ewolinetz) [inline roles/openshift_logging/tasks/oc_secret.yaml]: @kwoodson do we have a timeline on when the oc_secret module will be ported? I didn't have access to the trello card where this work is being tracked...

**Comment by ewolinetz** [comment]
Other than question regarding module usage over new task, LGTM

**Comment by kwoodson** [comment]
@ewolinetz,

oc_secret should be able to be finished today.  Here is the PR https://github.com/openshift/openshift-ansible/pull/3159

File: roles/openshift_logging/tasks/oc_secret.yaml (hunk 1 above)
Comment (ewolinetz) [inline roles/openshift_logging/tasks/oc_secret.yaml]: @jcantrill  looks like `oc_secret` was merged. can we try to use that module instead of this task?

**Comment by jcantrill** [comment]
@ewolinetz Looking at the referenced PR, I don't see where it provides options to 'link' a serviceaccount to a secret

**Comment by kwoodson** [comment]
@jcantrill,

We have another module named oc_secret_add that does the linking of secrets to serviceaccounts. It should be coming very soon.

**Comment by jcantrill** [comment]
@ewolinetz any opinions on merging this given we dont have an alternative right now?  Will ops need to specify pull secrets before we can utilize the module?

File: roles/openshift_logging/tasks/install_logging.yaml [hunk 2]
```diff
@@ -51,6 +51,26 @@
     loop_var: file
   when: ansible_check_mode
 
+- name: Link Pull Secrets With Service Accounts
```
Comment (ewolinetz) [inline roles/openshift_logging/tasks/install_logging.yaml]: It's pretty inconvenient that we aren't creating service accounts with a module meaning we need to add a pull secret out here...  can you add a TODO comment for this just as a reminder so we don't lose track of that? I'd like to make sure that when we have a module to create a service account idempotently that we would create a pull secret shortly after if applicable.

**Comment by ewolinetz** [comment]
nit and then LGTM I suppose. We'll need to refactor this with modules when they become available

File: roles/openshift_logging/tasks/install_logging.yaml (hunk 2 above)
Comment (jcantrill) [inline roles/openshift_logging/tasks/install_logging.yaml]: added TODO

**Comment by jcantrill** [comment]
@ewolinetz updated and rebased

**Comment by jcantrill** [comment]
aos-ci-test

**Pull request merged**